### ISSUE TITLE
Use NoReturn to avoid "inconsistent-return-statements" false positives

### ISSUE THREAD

**Issue by tyralla** [PyCQA/pylint#4122]
This is a duplicate of #3188 but comes with a suggestion.

Maybe Pylint could use the "NoReturn" type hint to stop raising unnecessary "inconsistent-return-statements" warnings?

It would be convenient if the following would pass:

```
from typing import NoReturn


def error(value: float) -> NoReturn:
    raise TypeError(f"Cannot do math with type {type(value).__name__}")


def do_math(value: float) -> float:
    try:
        return value + 1.0
    except TypeError:
        error(value)
```
